src/api/app/controllers/source_controller.rb

```diff
@@ -259,13 +258,13 @@ def show_project_pubkey
 
   # DELETE /source/:project/_pubkey
   def delete_project_pubkey
-    params[:user] = User.current.login
+    params[:user] = User.session!.login
     path = pubkey_path
 
     # check for permissions
     upper_project = @prj.name.gsub(/:[^:]*$/, '')
     while upper_project != @prj.name && upper_project.present?
-      if Project.exists_by_name(upper_project) && User.current.can_modify?(Project.get_by_name(upper_project))
+      if Project.exists_by_name(upper_project) && User.session!.can_modify?(Project.get_by_name(upper_project))
```

Comment:
The line 226 in this file is now `User.possibly_nobody.can_modify?(...)`, while here it is `User.session!.can_modify?(...)`. Shouldn't they both use `possibly_nobody`? Also for other lines where we call `.can_modify?`?

Reply:
In source controller all actions require login, so using possibly_nobody is indeed a smell. I reviewed that controller in detail and indeed all were wrong. Thanks for spotting